pymc4/distributions/tensorflow/continuous.py

```diff
@@ -0,0 +1,97 @@
+"""
+PyMC4 continuous random variables.
+
+Wraps selected tfp.distributions (listed in __all__) as pm.RandomVariables.
+Implements random variables not supported by tfp as distributions.
+"""
+import tensorflow_probability as tfp
+tfd = tfp.distributions
+
+
+from .distribution import ContinuousDistribution
+
+__all__ = ['Normal']
+
+
+class Normal(ContinuousDistribution):
+    r"""
+    Univariate normal random variable.
+
+    The pdf of this distribution is
+
+    .. math::
+
+       f(x \mid \mu, \tau) =
+           \sqrt{\frac{\tau}{2\pi}}
+           \exp\left\{ -\frac{\tau}{2} (x-\mu)^2 \right\}
+
+    Normal distribution can be parameterized either in terms of precision
+    or standard deviation. The link between the two parametrizations is
+    given by
+
+    .. math::
+
+       \tau = \dfrac{1}{\sigma^2}
+
+    .. plot::
+
+        import matplotlib.pyplot as plt
+        import numpy as np
+        import scipy.stats as st
+        plt.style.use('seaborn-darkgrid')
+        x = np.linspace(-5, 5, 1000)
+        mus = [0., 0., 0., -2.]
+        sigmas = [0.4, 1., 2., 0.4]
+        for mu, sigma in zip(mus, sigmas):
+            pdf = st.norm.pdf(x, mu, sigma)
+            plt.plot(x, pdf, label=r'$\mu$ = {}, $\sigma$ = {}'.format(mu, sigma))
+        plt.xlabel('x', fontsize=12)
+        plt.ylabel('f(x)', fontsize=12)
+        plt.legend(loc=1)
+        plt.show()
+
+    ========  ==========================================
+    Support   :math:`x \in \mathbb{R}`
+    Mean      :math:`\mu`
+    Variance  :math:`\dfrac{1}{\tau}` or :math:`\sigma^2`
+    ========  ==========================================
+
+    Parameters
+    ----------
+    mu : float
+        Mean.
+    sigma : float
+        Standard deviation (sigma > 0).
+
+    Examples
+    --------
+    .. code-block:: python
+        @pm.model
+        def model():
+            x = pm.Normal('x', mu=0, sigma=10)
+
+    Developer Notes
+    ---------------
+    Parameter mappings to TensorFlow Probability are as follows:
+
+    - mu: loc
+    - sigma: scale
+    """
+
+    def __init__(self, name, mu, sigma, **kwargs):
+        self.mu = mu
+        self.sigma = sigma
+        super().__init__(name, **kwargs)
+        self._init_backend()
+
+    def _init_backend(self):
+        self._backend_dist = tfd.Normal(
+            loc=self.mu, scale=self.sigma, name=self.name)
+
+    def sample(self, shape=(), seed=None):
+        return self._backend_dist.sample(shape, seed).numpy()
```

Comment:
That should not be converted to Numpy as we will use this within the backend. Instead, we can add `sample_numpy` that does all for conversion (should be possible in base class).

Reply:
Better move sample and log_prob into a base class for backended distributions
```python
class BackendDistribution(Distribution):
    def sample(self, ...):
        ...
    def log_prob(self, ...):
        ...

class BackendNormal(BackendDistribution, Normal):
    def _init_backend(self, ...):
        ...
```
This compositional inheritance will affect the method resolution order. If in eg you override `__init__` in BackendDistribution (this should not be the case), it will be resolved to there, but as you only override `sample`, `logprob` to interact with backend distribution and `_init_backend` is defined right here, you resolve

* `__init__` -> Normal
* `_init_backend` -> BackendNormal
* `sample`, etc -> BackendDistribution

The formal method resolution order is gonna be like this `BackendNormal->BackendDistribution->Normal->Distribution->Model`

Reply:
`BackendDistribution` added and `sample` and `log_prob` moved there. You are right, it should cover most of the cases. In the case when it doesn't, we can simply override it.